src/v2.rs

```diff
@@ -76,12 +76,12 @@ impl EszipV2 {
   /// header section of the eszip has been parsed. Once this function returns,
   /// the data section will not necessarially have been parsed yet. To parse
   /// the data section, poll/await the future returned in the second tuple slot.
-  pub async fn parse<R: tokio::io::AsyncRead + Unpin>(
-    mut reader: tokio::io::BufReader<R>,
+  pub async fn parse<R: futures::io::AsyncRead + Unpin>(
```

Comment:
@magurotuna any specific reason to convert `tokio::io` to `futures::io`? Structs that implement tokio::io::AsyncRead can no longer be used with parse() call.

Reply:
This is from the PR description:

> to replace tokio with futures in large parts of the code, because tokio has dropped support for the fs feature for wasm target since v1.21.0 (see https://github.com/tokio-rs/tokio/releases/tag/tokio-1.21.0). Switched to futures because it provides [futures::io::AllowStdIo](https://docs.rs/futures/latest/futures/io/struct.AllowStdIo.html) which is a good adapter that converts std stuff to AsyncRead, allowing us to use filesystem things for wasm build.

I think for almost all cases there should be something equivalent to tokio's stuff in futures. Do you have any practical issue in mind?

Reply:
I couldn't find something equivalent to https://docs.rs/tokio-util/latest/tokio_util/io/struct.StreamReader.html in futures crate. Do you have any thoughts?

Reply:
So you are looking for an adaptor that can turn `futures::future::Stream` into `futures::io::AsyncRead` right? If so, `into_async_read` should help https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#method.into_async_read